include/boost/spirit/home/x3/support/context.hpp

```diff
@@ -79,14 +93,16 @@ namespace boost { namespace spirit { namespace x3
     {
         template <typename ID, typename T, typename Next, typename FoundVal>
         inline Next const&
-        make_unique_context(T& /* val */, Next const& next, FoundVal&)
+        make_unique_context(T& /* val */, Next const& next, FoundVal&fv)
+        //x3::get<ID>(next)==fv; so return old context
         {
             return next;
         }
-        
+
         template <typename ID, typename T, typename Next>
         inline context<ID, T, Next>
         make_unique_context(T& val, Next const& next, unused_type)
+        //ID not found in next; so append val to existing context.
```

Comment:
If it is comment to function overload, please place them before template keyword.

Reply:
To understand how make_unique_context works, the code reader
1st has to notice the difference in the last argument for
the 2 make_unique_context overloads.  So the comment makes
more sense *after* seeing the difference in the last
argument; hence, the comment comes after the function
signature instead of before.